src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -138,6 +140,24 @@ public void resetStartsAndStops() {
         assertEquals(Collections.emptyList(), removeTriggerProperty(emptyInput));
     }
 
+    @Issue("JENKINS-51290")
+    @Test
+    public void testPreviousBuildFailedHard() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+
+        // First we simulate a build that has failed so hard it has a null execution
+        p.setDefinition(new CpsFlowDefinition("echo 'Not doing anything'", true));
+        WorkflowRun run = r.buildAndAssertSuccess(p);
+        Field f = run.getClass().getDeclaredField("execution");
+        f.setAccessible(true);
+        f.set(run, null);
```

Comment:
Doesn't look very close to a "real world scenario" to me?

Reply:
Yeah, It's not.  The real world scenario generate the same null execution, but requires a hard-kill of the Pipeline or a failure within the CPS VM thread.

Reply:
You could easily emulate it by a SCMCpsFlowDefinition pointing to a repo without Pipeline. FileSystem SCM (or small copy-paste of it) allows doing it without actual repos